spring-kafka/src/main/java/org/springframework/kafka/listener/AbstractConsumerSeekAware.java

```diff
@@ -54,7 +63,8 @@ public void onPartitionsAssigned(Map<TopicPartition, Long> assignments, Consumer
 		ConsumerSeekCallback threadCallback = this.callbackForThread.get(Thread.currentThread());
 		if (threadCallback != null) {
 			assignments.keySet().forEach(tp -> {
-				this.callbacks.put(tp, threadCallback);
+				this.callbacks.put(tp, threadCallback); // TODO: This should be removed when the existing `callbacks` field disappears.
```

Comment:
Therefore no need in this TODO if you just remove that `callbacks` in favor of `topicToCallbacks`.